components/datetime/src/pattern/item/ule.rs

```diff
@@ -0,0 +1,95 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::PatternItem;
+use crate::fields;
+use core::convert::TryFrom;
+use zerovec::ule::{AsULE, EqULE, ULE};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct PatternItemULE(pub [u8; 3]);
```

Comment:
The inner data should not be `pub` because you have invariants that you need to enforce